File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022w/ID07IO1/Bridge/src/ShapeFactory.java

import java.util.HashMap;
import java.util.List;
import java.util.stream.Collectors;

public abstract class ShapeFactory
{
   private static interface ShapeMaker
   {
      public Shape make(String text);
   }
   private static HashMap<String,ShapeMaker> map=new HashMap<String,ShapeMaker>()
   {
      {
         put("Elipsa",(text) -> new ShapeCircle(text));
         put("Prostokąt",(text) -> new ShapeRectangle(text));
         put("Trójkąt",(text) -> new ShapeTriangle(text));
      }
   };
   private static List<String> keys=map.keySet().stream().sorted().collect(Collectors.toList());
   public static List<String> keySet() { return keys; }
   public static Shape make(String kind,String text) { return map.get(kind).make(text); }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO2/BridgePatern/BridgePatern/ShapeFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePatern
{
    public class ShapeFactory
    {
        private delegate Shape ShapeMaker(string parameters);
        private static Dictionary<string, ShapeMaker> map = new Dictionary<string, ShapeMaker>()
        {
            {"Circle",    (parameters) => new ShapeCircle(parameters)    },
            {"Rectangle", (parameters) => new ShapeRectangle(parameters) },
            {"Triangle",  (parameters) => new ShapeTriangle(parameters)  },
        };
        public static string[] Kinds = map.Keys.OrderBy(kind => kind).ToArray();
        public static Shape make(string kind, string parameters)
        {
            return map[kind](parameters);
        }
    }
}
